conda/gateways/disk/delete.py

```diff
@@ -68,25 +72,35 @@ def unlink_or_rename_to_trash(path):
         make_writable(path)
         unlink(path)
     except EnvironmentError:
-        try:
-            rename(path, path + ".trash")
-        except EnvironmentError:
-            if on_win:
-                # on windows, it is important to use the rename program, as just using python's
-                #    rename leads to permission errors when files are in use.
-                condabin_dir = join(context.conda_prefix, "condabin")
-                trash_script = join(condabin_dir, 'rename_tmp.bat')
-                if exists(trash_script):
-                    _dirname, _fn = split(path)
-                    p = Popen(['cmd.exe', '/C', trash_script, _dirname, _fn, _fn + ".trash"],
-                              stdout=PIPE, stderr=PIPE)
-                    stdout, stderr = p.communicate()
-                else:
-                    log.debug("{} is missing.  Conda was not installed correctly or has been "
-                              "corrupted.  Please file an issue on the conda github repo."
-                              .format(trash_script))
-            log.warn("Could not remove or rename {}.  Please remove this file manually (you may "
-                     "need to reboot to free file handles)".format(path))
+        if not fnmatch.fnmatch(path, '*.conda_trash*'):
+            try:
+                rename(path, path + ".conda_trash")
+            except EnvironmentError:
+                if on_win:
+                    # on windows, it is important to use the rename program, as just using python's
+                    #    rename leads to permission errors when files are in use.
+                    condabin_dir = join(context.conda_prefix, "condabin")
+                    trash_script = join(condabin_dir, 'rename_tmp.bat')
+                    if exists(trash_script):
+                        _dirname, _fn = split(path)
+                        dest_fn = path + ".conda_trash"
+                        counter = 1
```

Comment:
What's the significance of this counter variable? Is it supposed to be incremented somewhere?

Reply:
it wasn't being incremented - that was a bug.  Thanks.  It is used to find the first non-existing .conda_trash filename suffix that's available.